enhancements/ingress/gateway-api-crd-life-cycle-management.md

```diff
@@ -0,0 +1,437 @@
+disruptions to existing workloads.
+
+#### Using a third-party Gateway API implementation
+
+As a cluster-admin, I want to install a third-party Gateway API implementation
+on my OpenShift 4.19 cluster, and use the third-party implementation without
+any interference from the first-party implementation. Relatedly, I want to be
+able to utilize both the first-party and any third-party solution alongside
+each other simultaneously and independently without any interference between the
+two.
+
+#### Future OpenShift upgrades
+
+As a cluster-admin, I want to be able to receive updates to the Gateway API
+resources via zstream (and major) releases to add new features and capabilities.
+
+### Goals
+
+- Ensure Gateway API CRDs are installed on new and upgraded OpenShift 4.19 clusters.
+- Ensure the installed Gateway API CRDs are compatible with OpenShift's needs.
+- Protect against bad updates or removals of the Gateway API CRDs.
+- Detect and warn if an incompatible version of the CRDs is installed.
+- Protect users against "unknown fields"/"dead fields" (defined below).
+- Provide a method to transfer ownership of previously existing CRDs to when upgrading from 4.18 to 4.19.
+
+### Non-Goals
+
+- _Automatically_ replace incompatible CRDs that some other agent installed.
+- Provide an explicit override for the cluster-admin to take CRD ownership.
+- Solve CRD life-cycle management in OLM.
+- Solve OLM subscription management.
+
+## Proposal
+
+### CRD Life-cycle management
+
+OpenShift's [Cluster Ingress Operator (CIO)] will manage the _entire_
+life-cycle of the Gateway API CRDs from here onward. The CIO will be packaged
+along with a [Validating Admission Policy (VAP)] to block updates from sources
+other than the CIO, and will hold the CRDs at a specific `$CRD_VERSION`. In
+effect, this means that the **Gateway API resources will now be treated like a
+core API.**
+
+[Cluster Ingress Operator (CIO)]:https://github.com/openshift/cluster-ingress-operator
+[Validating Admission Policy (VAP)]:https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/
+
+#### CRD Deployment
+
+The CIO will now check for the presence of the CRDs on the cluster, for
+which there are two scenarios:
+
+1. The CRDs are not present, so we create them
+2. The CRDs are already present, so we need to take over management
+
+The former is extremely straightforward and the correct version of the CRDs
+will be applied.
+
+The latter situation has more complexities. We'll refer to this process as "CRD
+Management Succession", and cover its implications and logic below.
+
+#### CRD Management Succession
+
+Taking over the management of the Gateway API CRDs from a previous entity
+can be tumultuous as we will require:
+
+* no other actors managing the CRDs from here on out
+* only standard CRDs can be deployed (no experimental)
+* the CRDs to be deployed at an exact version/schema
+* OR the CRDs not to be present at all
```

Comment:
This will read a little better with proper grammatical parallelism and punctuation: 
```suggestion
can be tumultuous as we will require the following:

* no other actors to be managing the CRDs from here on out,
* AND only standard CRDs to be deployed (no experimental),
* AND the CRDs to be deployed at an exact version/schema;
* OR the CRDs not to be present at all.
```